### DISCUSSION TITLE
What's in a good Helm chart?

### DISCUSSION THREAD

**Discussion by rfratto** [grafana-cold-storage/agent#738] [General]
We're planning on building an official Helm chart for the Grafana Agent Operator once we complete support for traces, logs, and integrations. 

I'm really interested in collecting community feedback on:

* What do you think a "good" Helm chart is?
* Examples of your favorite Helm charts
* For Helm charts that deploy an operator, does it deploy any custom resources? How do you deploy custom resources to your cluster? Normal `kubectl apply`? 

If you don't have any answers to this questions, I'd also love to hear in general what you'd like to see from the Grafana Agent Operator Helm chart!

**Comment by rverma-dev** [comment]
Istio ones are pretty neat, have a look at https://github.com/istio/istio/tree/master/manifests/charts/istio-operator. Even I like the current grafana ones, apart from cortex. One thing is that the examples/libsonnet are there with raw configs, and in usual cases I would prefer to consume in the same way. 
For initial days, its a good idea to provide something with couple of parameters exposed through `values.yaml` but I usually use that to view the generated config file and then copy it over.

**Comment by hjet** [comment]
I will try to add more later today, but for the 3rd bullet (CRDs), [kube-prometheus](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack) uses the `crds` special dir that Helm installs first (see https://helm.sh/docs/chart_best_practices/custom_resource_definitions/)

**Reply by hjet** [reply]
prom operator (https://github.com/helm/charts/tree/master/stable/prometheus-operator) and jaeger (https://github.com/jaegertracing/helm-charts/tree/main/charts/jaeger-operator) do this too

**Reply by hjet** [reply]
however, if you want to go down the rabbit hole: see https://github.com/helm/helm/issues/5871 and https://github.com/helm/helm/issues/7735
